File: packages/http-client-csharp-generator/src/AutoRest.CSharp/Common/Input/InputTypes/InputEnumTypeValue.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

namespace AutoRest.CSharp.Common.Input;

internal record InputEnumTypeValue(string Name, object Value, string? Description)
{
    public virtual string GetJsonValueString() => GetValueString();
    public string GetValueString() => (Value.ToString() ?? string.Empty);
}

File: packages/http-client-csharp/emitter/src/type/input-enum-type-value.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import { DecoratorInfo } from "@azure-tools/typespec-client-generator-core";

export interface InputEnumTypeValue {
  Name: string;
  Value: any;
  Description?: string;
  Decorators?: DecoratorInfo[];
}
